silo/views.py

```diff
@@ -437,6 +437,23 @@ def editSilo(request, id):
     :param id: Unique table ID
     :return: silo edit form
     """
+
+    edited_silo = Silo.objects.get(pk=id)
+
+    request_user_org=None
+    owner_user_org=None
+    if(hasattr(request.user, 'tola_user') and
+            hasattr(edited_silo.owner,'tola_user')):
+        request_user_org = request.user.tola_user.organization
+        owner_user_org = edited_silo.owner.tola_user.organization
+
+    if not (edited_silo.owner == request.user or edited_silo.public
+            or request.user in edited_silo.shared.all()
```

Comment:
It better to use the `exists` method instead of fetching the whole data from the db.